tracing-opentelemetry/src/layer.rs

```diff
@@ -1386,4 +1477,184 @@ mod tests {
             ]))
         );
     }
+
+    #[test]
+    fn tracks_dropped_events() {
+        let tracer = TestTracer(Arc::new(Mutex::new(None)));
+        let subscriber = tracing_subscriber::registry()
+            .with(layer().with_tracer(tracer.clone()).max_events_per_span(4));
+        let subscriber = Arc::new(subscriber);
+
+        tracing::subscriber::with_default(subscriber.clone(), || {
+            let span = tracing::debug_span!("request");
+            let _span_guard = span.enter();
+
+            // we should record all 4 events
+            tracing::info!("event 1");
+            tracing::info!("event 2");
+            tracing::info!("event 3");
+            tracing::info!("event 4");
+
+            let data = span.id().and_then(|id| subscriber.span(&id)).unwrap();
+            let extensions = data.extensions();
+
+            // we shouldn't have recoreded any dropped events
+            assert!(extensions.get::<DroppedOtelEvents>().is_none());
```

Comment:
can you also just quickly test this in one of the other tests that doesnt set `max_events_per_span`?

Reply:
good idea! I added another test `exclude_max_events_per_span` which seems to follow the convention of the other `exclude_*` tests